native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -228,6 +219,43 @@ private static Object nativeQueryRowExecutable(
         return ErrorGenerator.getSQLApplicationError("Client is not properly initialized!");
     }
 
+    private static Object getUnionTypeBValue(
+            UnionType describingType, ResultSet resultSet, AbstractResultParameterProcessor resultParameterProcessor)
+            throws SQLException, TypeMismatchError {
+        for (Type type: describingType.getMemberTypes()) {
+            try {
+                // If one of the types inside the union is a union, recursively check
+                if (type.getTag() == TypeTags.UNION_TAG) {
+                    return getUnionTypeBValue(describingType, resultSet, resultParameterProcessor);
```

Comment:
Let's add a test for this too. Union type of 3 types.

Reply:
Test for this is included in https://github.com/ballerina-platform/module-ballerina-sql/pull/426/files#diff-5c3e102511b7f2f9e73703f8686cf12f5d665a855281f2f35c0111f3fc3b052dR1107